### PR TITLE
Add CoulombMat calculator

### PR DESCRIPTION
**Pull request by thegodone**
#### What does this implement/fix? Explain your changes.
This is a addition feature implementing Coulomb Matrix computation based on 3D molecule structure (severals parameters to be able to generate global or local representations including sorted/decayed/reduced forms)

### CODE REVIEW

**Comment by greglandrum** [comment]
@ptosco : since I helped out with writing this one, would you mind reviewing it before we merge?

**Comment by ptosco** [comment]
@greglandrum sure, will do tomorrow.

File: Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp [hunk 1]
```diff
@@ -118,6 +118,16 @@ python::tuple calcCrippenDescriptors(const RDKit::ROMol &mol,
 
 #ifdef RDK_BUILD_DESCRIPTORS3D
 
+python::tuple calcCoulombMat(RDKit::ROMol &mol, int confId) {
```
Comment (ptosco) [inline Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp]: I'd change this into
```
python::tuple calcCoulombMat(const RDKit::ROMol &mol, int confId)
```

File: Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp [hunk 2]
```diff
@@ -1548,6 +1556,13 @@ BOOST_PYTHON_MODULE(rdMolDescriptors) {
               python::return_value_policy<python::manage_new_object>());
 
 #ifdef RDK_BUILD_DESCRIPTORS3D
+  python::scope().attr("_CalcCoulombMat_version") =
+      RDKit::Descriptors::CoulombMatVersion;
+  docString = "Returns severals Coulomb randomized Matrixes";
```
Comment (ptosco) [inline Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp]: Change `Matrixes` into `matrices`.

File: Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp [hunk 3]
```diff
@@ -1688,44 +1703,47 @@ BOOST_PYTHON_MODULE(rdMolDescriptors) {
...
 
   std_pair_to_python_converter<double, double>();
-  docString = \
-    "Returns a 2D BCUT (eigen value hi, eigenvalue low) given the molecule and the specified atom props\n"\
-    " there length ot atom_props must a list or tuple of floats equal in size to the number of atoms in mol";
-  
+  docString =
+      "Returns a 2D BCUT (eigen value hi, eigenvalue low) given the molecule "
+      "and the specified atom props\n"
+      " there length ot atom_props must a list or tuple of floats equal in "
```
Comment (ptosco) [inline Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp]: Change
```
 there length ot atom_props must a list
```
into
```
atom_props must be a list
```

File: Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp [hunk 4]
```diff
@@ -1688,44 +1703,47 @@ BOOST_PYTHON_MODULE(rdMolDescriptors) {
...
               (python::arg("mol"), python::arg("atom_props")),
               docString.c_str());
 
-  docString = \
-    "Returns a 2D BCUT (eigen value high, eigen value low) given the molecule and the specified atom prop name\n" \
-    "atom_propname must exist on each aton and be convertable to a float";
+  docString =
+      "Returns a 2D BCUT (eigen value high, eigen value low) given the "
+      "molecule and the specified atom prop name\n"
+      "atom_propname must exist on each aton and be convertable to a float";
```
Comment (ptosco) [inline Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp]: Change
```
aton and be convertable
```
into
```
atom and be convertible
```

**Comment by ptosco** [comment]
@greglandrum OK to merge for me.

**Pull request merged**